allenact_plugins/manipulathor_plugin/manipulathor_environment.py

```diff
@@ -0,0 +1,407 @@
+"""A wrapper for engaging with the ManipulaTHOR environment."""
+
+import copy
+import math
+import warnings
+from typing import Tuple, Dict, List, Set, Union, Any, Optional, cast
+
+import ai2thor.server
+import numpy as np
+from ai2thor.controller import Controller
+from allenact_plugins.ithor_plugin.ithor_constants import VISIBILITY_DISTANCE, FOV
+from allenact_plugins.ithor_plugin.ithor_environment import IThorEnvironment
+
+from allenact_plugins.manipulathor_plugin.manipulathor_constants import (
+    ADITIONAL_ARM_ARGS,
+    ARM_MIN_HEIGHT,
+    ARM_MAX_HEIGHT,
+    MOVE_ARM_HEIGHT_CONSTANT,
+    MOVE_ARM_CONSTANT,
+    reset_environment_and_additional_commands,
+    MOVE_THR,
+)
+
+
+class ManipulaTHOREnvironment(IThorEnvironment):
+    """Wrapper for the manipulathor controller providing arm functionality
+    and bookkeeping.
+
+    See [here](https://ai2thor.allenai.org/documentation/installation) for comprehensive
+     documentation on AI2-THOR.
+
+    # Attributes
+
+    controller : The ai2thor controller.
+    """
+
+    def __init__(
+            self,
```

Comment:
This spacing suggests to me that black hasn't been run on this file before committing. Can you call `bash scripts/auto_format.sh` from top level directory of the repository?

Reply:
Done